analyzers/tests/SonarAnalyzer.UnitTest/TestCases/DoNotCheckZeroSizeCollection.cs

```diff
@@ -44,28 +45,31 @@ public void TestCountMethod()
 
             var someEnumerable = new List<string>();
 
-            result = someEnumerable.Count() >= 0; // Noncompliant {{The 'Count' of 'IEnumerable<T>' is always '>=0', so fix this test to get the real expected behavior.}}
-//                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^
+            result = someEnumerable.Count() >= 0; // Noncompliant {{The 'Count' of 'IEnumerable<T>' always evaluates as 'True' regardless the size.}}
+            //       ^^^^^^^^^^^^^^^^^^^^^^^^^^^
             result = someEnumerable.Count(foo => true) >= 0; // Noncompliant
             result = someEnumerable?.Count() >= 0; // Noncompliant
 
             result = someEnumerable.Count() >= 1;
             result = someEnumerable.Count() >= localVariable;
-            result = someEnumerable.Count() >= -1; // Noncompliant {{The 'Count' of 'IEnumerable<T>' is always '>=0', so fix this test to get the real expected behavior.}}
+            result = someEnumerable.Count() >= -1; // Noncompliant {{The 'Count' of 'IEnumerable<T>' always evaluates as 'True' regardless the size.}}
             result = someEnumerable.Count() <= 0;
-            result = someEnumerable.Count() < 0; // Noncompliant
+            result = someEnumerable.Count() < 0; // Noncompliant {{The 'Count' of 'IEnumerable<T>' always evaluates as 'False' regardless the size.}}
             result = 0 >= someEnumerable.Count();
 
             result = someEnumerable.Count() >= localConst_Zero; // Noncompliant
             result = someEnumerable.Count() >= ConstField_NonZero;
             result = someEnumerable.Count() >= ConstField_Zero; // Noncompliant
             result = someEnumerable.Count() >= localConst_NonZero;
+            result = someEnumerable.Count() >= ConstField_Double_Zero; // Compliant, double is ignored
 
             result = (someEnumerable.Count()) >= (0); // Noncompliant
             result = ((((someEnumerable).Count())) >= ((0))); // Noncompliant
-//                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+            //        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
             result = 0 <= someEnumerable.Count(); // Noncompliant
-//                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^
+            //       ^^^^^^^^^^^^^^^^^^^^^^^^^^^
+            
+            result = result.ToString() == "Not integer";            
```

Comment:
IT threw this exception, please add this UT

>AD0001: Analyzer 'SonarAnalyzer.Rules.CSharp.DoNotCheckZeroSizeCollection' threw an exception of type 'System.InvalidOperationException' with message 'Unexpected node: AssignmentExpressionSyntax'.

```suggestion
            result = result.ToString() == "Not integer";            
            result = localVariable = 42 == 0;
            result = 0 == localVariable = 42;

```